### PR TITLE
fix: ignore security requirements when unsuccessful

### PR DESCRIPTION
**Pull request by vwong**

### CODE REVIEW

**Review by mefellows — CHANGES_REQUESTED** [review]
Thanks Voon

Could you please add some tests to provide clarity that we also support

- Bad requests (40x - e.g. missing headers, query, invalid paths, bodies etc.)
- Not found requests (404)

In the above cases, a user should be able to provide invalid requests, and have the tests pass if the status code matches a defined response in the OAS.

As discussed, looking at this PR would make it look like we only support negative scenarios for security related errors.

File: test/unit/response-body.spec.ts [hunk 1]
```diff
@@ -1,676 +1,714 @@
...
+import { operationBuilder } from './support/swagger2-builder/operation-builder';
+import { pathBuilder } from './support/swagger2-builder/path-builder';
+import { responseBuilder } from './support/swagger2-builder/response-builder';
+import { schemaBuilder, SchemaBuilder } from './support/swagger2-builder/schema-builder';
 
 declare function expect<T>(actual: T): CustomMatchers<T>;
 
-describe('response body', () => {
-    const expectedFailedValidationError = 'Mock file "pact.json" is not compatible with spec file "spec.json"';
+[200, 401, 404].forEach((statusCode) => {
```
Comment (vwong) [inline test/unit/response-body.spec.ts:13]: The real change here is to add multiple `statusCode`, and to show that the response bodies from each of these are checked

**Review by vwong — COMMENTED** [review]
Added/extended `shouldSkipValidation` in a few key scenarios:
* request params (which includes routes)
* request headers
* request body
* request security

A general 404 can't be supported because the OAS doesn't have one (or does it?)

**Comment by mefellows** [comment]
> A general 404 can't be supported because the OAS doesn't have one (or does it?)

Are you saying that in the case a user doesn't define a `404` in the OAS, but expects one in the Pact it would fail the check? If so, that is sensible to me. We should only allow it to pass where an explicit status code exists in the OAS, I think.

**Comment by vwong** [comment]
As I understand OAS, you can attach responses to specific routes only, there is no catch-all route. So, if there is a 404 response for a given route, then we'll match it accordingly, but we can't catch-all 404s.

**Review by mefellows — APPROVED** [review]

**Pull request merged**